activesupport/lib/active_support/core_ext/date_and_time/calculations.rb

```diff
@@ -163,8 +163,17 @@ def end_of_quarter
     #   Date.new(2010, 4, 12).quarter  # => 2
     #   Date.new(2010, 9, 15).quarter  # => 3
     #   Date.new(2010, 12, 25).quarter # => 4
-    def quarter
-      (month / 3.0).ceil
+    #
+    #   Date.new(2010, 1, 31).quarter(fiscal: true)  # => 4
+    #   Date.new(2010, 4, 12).quarter(fiscal: true)  # => 1
+    #   Date.new(2010, 9, 15).quarter(fiscal: true)  # => 2
+    #   Date.new(2010, 12, 25).quarter(fiscal: true) # => 3
+    def quarter(fiscal: false)
+      if fiscal
+        (advance(months: -3).month / 3.0).ceil
```

Comment:
I'm guessing you picked April 1st as the start date of a fiscal year because you're based in India according to your profile? That doesn't work worldwide, it is October 1st for the US government, June 1st for Microsoft, and February 1st for Alphabet to name a few.

Reply:
In that case, should we add an option to accept the `start_month` of the financial year?

We can calculate the quarter based on that.

```ruby
class Date
  def quarter(fiscal_start: 1)
    (advance(months: -(fiscal_start - 1)).month / 3.0).ceil
  end
end
```

```
fiscal_start: 4
2022-01-20 => 4
2022-02-20 => 4
2022-03-20 => 4
2022-04-20 => 1
2022-05-20 => 1
2022-06-20 => 1
2022-07-20 => 2
2022-08-20 => 2
2022-09-20 => 2
2022-10-20 => 3
2022-11-20 => 3
2022-12-20 => 3

=====
fiscal_start: 7
2022-01-20 => 3
2022-02-20 => 3
2022-03-20 => 3
2022-04-20 => 4
2022-05-20 => 4
2022-06-20 => 4
2022-07-20 => 1
2022-08-20 => 1
2022-09-20 => 1
2022-10-20 => 2
2022-11-20 => 2
2022-12-20 => 2
```

That way everyone's happy
Thoughts? 

@bdewater @matthewd

Reply:
The key phrase there was "we specifically chose".